packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js

```diff
@@ -304,6 +289,21 @@ export const BlocksDropdown = ({ disabled }) => {
     }
   };
 
+  React.useEffect(() => {
+    if (editor.selection) {
+      const [anchorNode] = Editor.parent(editor, editor.selection.anchor); // Get the parent node of the anchor
+      const anchorBlockKey = Object.keys(blocks).find((blockKey) =>
+        blocks[blockKey].matchNode(anchorNode)
+      );
+
+      if (anchorNode) {
+        if (anchorBlockKey && anchorBlockKey !== blockSelected) {
+          setBlockSelected(anchorBlockKey);
+        }
+      }
+    }
+  }, [editor.selection, editor, blocks, blockSelected]);
```

Comment:
This needs a comment as to _why_ you have to do it, people will not understand just from reading this because its very unclear

Reply:
yes, sorry, I provided some comments to explain line by line the useEffect